latch_cli/services/register/register.py

```diff
@@ -212,21 +225,73 @@ def _build_and_serialize(
             jit_wf,
             ctx.pkg_root,
             ctx.snakefile,
-            ctx.version,
+            ctx.dkr_repo,
             image_name,
             current_workspace(),
         )
 
-    image_build_logs = build_image(ctx, image_name, context_path, dockerfile)
+    image_build_logs = build_image(
+        ctx, image_name, context_path, dockerfile, cache=cache
+    )
     print_and_write_build_logs(
         image_build_logs, image_name, ctx.pkg_root, progress_plain=progress_plain
     )
 
     if ctx.workflow_type == WorkflowType.snakemake:
         assert jit_wf is not None
         assert ctx.dkr_repo is not None
+        assert ctx.snakefile is not None
 
         from ...snakemake.serialize import serialize_jit_register_workflow
+        from ...snakemake.wrappers.build import get_dockerfile_content
+        from ...snakemake.wrappers.extractor import SnakemakeWrapperExtractor
+
+        wrapper_extractor = SnakemakeWrapperExtractor(
+            pkg_root=ctx.pkg_root, snakefile=ctx.snakefile
+        )
+        wrapper_extractor.include(ctx.snakefile)
+        wrappers = wrapper_extractor.extract_wrappers()
+
+        if build_wrappers:
+            for i, wrapper in enumerate(wrappers):
+                suffix = identifier_suffix_from_str(wrapper)
+                wrapped_image = f"{ctx.image}_{suffix}"
+                wrapped_tagged = f"{wrapped_image}:{ctx.version}"
+
+                _docker_login(ctx, override_image=wrapped_image)
+
+                with tempfile.TemporaryDirectory() as td:
+                    dockerfile = Path(td) / "Dockerfile"
+                    dockerfile.write_text(get_dockerfile_content(wrapper))
+
+                    click.echo()
+                    click.echo(
+                        "Building Wrapper"
+                        f" {click.style(wrapper, italic=True, underline=True)} [{i + 1}/{len(wrappers)}]"
+                    )
+
+                    build_logs = build_image(
+                        ctx,
+                        wrapped_tagged,
+                        context_path,
+                        dockerfile=dockerfile,
+                        login=False,
+                        cache=cache,
+                    )
+                    print_and_write_build_logs(
+                        build_logs,
+                        wrapped_tagged,
+                        ctx.pkg_root,
+                        progress_plain=progress_plain,
+                        print_build_message=False,
```

Comment:
`print_build_message=print_build_message`?

Reply:
`print_build_message` is not an argument passed to register - this is set to `False` only so that wrapper builds look fine in the terminal